crates/bevy_ecs/src/event.rs

```diff
@@ -202,14 +203,32 @@ impl<E: Event> Events<E> {
 
         self.events_b.push(event_instance);
         self.event_count += 1;
+
+        event_id
+    }
+
+    /// Sends a list of `events` all at once, which can later be read by [`EventReader`]s.
+    /// This is more efficient than sending each event individually.
+    /// This method returns the [IDs](`EventId`) of the sent `events`.
+    pub fn send_batch(
+        &mut self,
+        events: impl IntoIterator<Item = E>,
+    ) -> impl Iterator<Item = EventId<E>> {
```

Comment:
I would define my own iterator and return it instead of returning `impl Iterator`. This would also fix the issue of requring `let _ = …`

```rust
struct SendBatchIds<E> {
    last_count: usize,
    event_count: usize,
    _marker: PhantomData<E>,
}
impl<E> Iterator for SendBatchIds<E> {
    type Item = EventId<E>;

    fn next(&mut self) -> Option<Self::Item> {
        if self.last_count >= self.event_count {
            return None;
        }
        self.last_count += 1;
        Some(EventId {
            id: self.last_count - 1,
            _marker: PhatomData,
        })
    }
}
```

Reply:
That's a great idea! I've incorporated this suggestion almost verbatim, and you're right it does remove the `must_use` warning. Thanks!